.github/workflows/nuget-publish.yml

```
@@ -0,0 +1,55 @@
+name: Release Package Version
+on:
+  push:
+    tags:
+     - v*     
+jobs: 
+  build:
+    if: github.event.base_ref == 'refs/heads/main'
+    runs-on: ubuntu-latest
+    steps:    
+    
+    - name: Print Tag Ref
+      run: echo $GITHUB_REF
+      
+    - name: Extract Version Number
+      uses: actions-ecosystem/action-regex-match@v2
+      id: regex-match
+      with:
+        text: ${{ github.ref }}
+        regex: '[0-9.]+'
+       
+    - name: Print Version Number
+      run: echo '${{ steps.regex-match.outputs.match }}'
+      
+    - uses: actions/checkout@v2
+    - name: Setup .NET  
+      uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: 5.0.x
+        
+    - name: Restore dependencies
+      run: dotnet restore
+      
+    - name: Build
+      run: |
+        dotnet build --configuration Release --no-restore
+      
+    - name: Test
+      run: |
+        dotnet test --no-restore --verbosity normal
+      
+    - name: Pack
+      run: |
+        dotnet pack ./Microsoft.Azure.CosmosRepository/src/Microsoft.Azure.CosmosRepository.csproj -p:PackageVersion='${{ steps.regex-match.outputs.match }}' --output packages
```

```suggestion
        dotnet pack ./Microsoft.Azure.CosmosRepository/src/Microsoft.Azure.CosmosRepository.csproj --output packages
```

You shouldn't need to manually apply the version, this is handled implicitly on your behalf from MinVer based on the tag. See [here](https://rehansaeed.com/the-easiest-way-to-version-nuget-packages/).

---

Ahh okay I see that is pretty cool.